librz/il/rzil_opcodes.c

```diff
@@ -0,0 +1,176 @@
+// SPDX-FileCopyrightText: 2021 heersin <teablearcher@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_il/rzil_opcodes.h>
+
+#define RZIL_OP_NEW(size) calloc(1, (size))
+
+RzILOp rz_il_new_empty_op(void) {
+	RzILOp ret = (RzILOp)malloc(sizeof(struct RzILOp_t));
+	if (!ret) {
+		return NULL;
+	}
+	ret->id = 0;
+	ret->code = RZIL_OP_INVALID;
+	ret->op.nil = NULL;
+
+	return ret;
+}
+
+/**
+ * Create an empty core theory op, argument of the op should be set in analysis_[arch]
+ * \param code RzILOPCode, enum to specify the op type
+ * \return RzILOp, a pointer to an empty opcode instance
+ */
+RZ_API RzILOp rz_il_new_op(RzILOPCode code) {
+	RzILOp ret = (RzILOp)malloc(sizeof(struct RzILOp_t));
+	if (!ret) {
+		return NULL;
+	}
+	ret->id = 0;
+	ret->code = code;
+
+	switch (code) {
+	case RZIL_OP_VAR:
+		ret->op.var = (RzILOpVar)RZIL_OP_NEW(sizeof(struct rzil_op_var_t));
```

Comment:
```suggestion
		ret->op.var = RZ_NEW0(RzILOpVar);
```
this and all the others.

Reply:
but `RZ_NEW0(RzILOpVar)` will only allocate a 4-byte memory for `RzILOpVar`. (it's a pointer)